### PR TITLE
Python: Add spellcheck to the CI

### PR DESCRIPTION
**Pull request by Fokko**
Simple spellcheck so we don't have to spend any mental energy on the spelling

### CODE REVIEW

File: python/src/iceberg/table/partitioning.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ class PartitionField:
 
     Attributes:
         source_id(int): The source column id of table's schema
-        field_id(int): The partition field id across all the table metadata's partition specs
+        field_id(int): The partition field id across all the table partition metadata specs
```
Comment (dramaticlly) [inline python/src/iceberg/table/partitioning.py:27]: nit: I think [partition spec](https://iceberg.apache.org/spec/#partition-specs) here refer to a single entity in iceberg jargon, so I feel `partition metadata specs` does not confer the same meaning

File: python/spellcheck-dictionary.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+accessor
```
Comment (dramaticlly) [inline python/spellcheck-dictionary.txt:1]: just curious, do we need to manually maintain the private dictionary going forward?

Comment (dramaticlly) [reply python/spellcheck-dictionary.txt:1]: also I am curious, do we need both `python/spellcheck-dictionary.txt` and top level `spellcheck-dictionary.txt`?

File: python/src/iceberg/table/partitioning.py (hunk 1 above)
Comment (Fokko) [reply python/src/iceberg/table/partitioning.py:27]: I've updated it to `The partition field id across all the table partition specs metadata` WDYT?

File: python/spellcheck-dictionary.txt (hunk 2 above)
Comment (Fokko) [reply python/spellcheck-dictionary.txt:1]: > just curious, do we need to manually maintain the private dictionary going forward?

Yes, once there is special jargon that's outside of the English dictionary, we need to add this.

> also I am curious, do we need both python/spellcheck-dictionary.txt and top level spellcheck-dictionary.txt?

Yes, in that case, we would have another dictionary, but a better option would be to move it to the top level. Or in `dev/` to keep the top-level nice and clean.

File: dev/.rat-excludes [hunk 3]
```diff
@@ -25,4 +25,5 @@ package-list
 sitemap.xml
 derby.log
 .python-version
+spellcheck-dictionary.txt
```
Comment (rdblue) [inline dev/.rat-excludes:28]: There's no way to add comments to this file?

File: python/src/iceberg/table/partitioning.py [hunk 4]
```diff
@@ -24,7 +24,7 @@ class PartitionField:
 
     Attributes:
         source_id(int): The source column id of table's schema
-        field_id(int): The partition field id across all the table metadata's partition specs
+        field_id(int): The partition field id across all the table partition specs' metadata
```
Comment (rdblue) [inline python/src/iceberg/table/partitioning.py:27]: This changes the meaning. "Table metadata" refers to all of the metadata, and "table metadata's partition specs" is the set of partition specs in table metadata. "Partition specs' metadata" is narrower, the metadata across partition specs.

To fix this, I think you can just omit "metadata": "across all the table's partition specs".

File: python/tests/test_schema.py [hunk 5]
```diff
@@ -228,7 +228,7 @@ def test_schema_find_field_by_id_raise_on_unknown_field(table_schema_simple):
 
 
 def test_schema_find_field_type_by_id(table_schema_simple):
-    """Test retrieving a columns's type using its field ID"""
+    """Test retrieving a column type using its field ID"""
```
Comment (rdblue) [inline python/tests/test_schema.py:231]: "column's"? I think the possessive form should still be used.

File: python/tox.ini [hunk 6]
```diff
@@ -52,10 +52,13 @@ deps =
     black
     isort
     autoflake
+    pylint
+    pyenchant
 commands =
     autoflake -r --check --ignore-init-module-imports --remove-all-unused-imports src tests
     isort --profile black --check-only src tests
     black --line-length 130 --check --diff src tests
+    pylint --disable all --enable spelling --spelling-dict en_US --spelling-private-dict-file spellcheck-dictionary.txt en_US src tests
```
Comment (kbendick) [inline python/tox.ini:61]: Question / Nit: Do we need to specify `en_US` twice? Specifically for the 2nd argument after `--spelling-private-dict-file`.

Comment (Fokko) [reply python/tox.ini:61]: Great catch @kbendick! That's just me being messy with copy and paste!

File: python/tests/test_schema.py (hunk 5 above)
Comment (Fokko) [reply python/tests/test_schema.py:231]: Updated!

File: python/src/iceberg/table/partitioning.py (hunk 4 above)
Comment (Fokko) [reply python/src/iceberg/table/partitioning.py:27]: Love it, less is more 👍🏻

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @Fokko!

**Pull request merged**